### PR TITLE
better sdist api

### PR DESCRIPTION
**Pull request by dholth**
@magv I also format the code automatically with black. It eats the u"" from strings. Rather than figure out how to make it stop, I've changed _is_string(). I would also be open to using Python-Requires: to release the last Python 2 compatible version of enscons.

### CODE REVIEW

File: enscons/__init__.py [hunk 1]
```diff
@@ -194,9 +194,15 @@ def egg_info_builder(target, source, env):
             entry_points_builder([dnode], source, env)
 
 
+try:
+    basestring
+except NameError:
+    basestring = str
+
```
Comment (magv) [inline enscons/__init__.py:201]: Brutal :)

**Comment by magv** [comment]
The project I'm trying to convert to enscons may support Python 2 for one more release, but that is about it (and even that is a question). I say the next time Python 2 compatibility would require any mental effort to maintain, just drop it.

**Comment by dholth** [comment]
@magv You must have some nice complex builds

**Comment by magv** [comment]
Not particularly complex, just some C programs that must be shipped together with the Python code.

**Pull request merged**